include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -139,33 +139,35 @@ void hipLaunchKernelGGLImpl(
 } // Namespace hip_impl.
 
 
-template <typename F>
+template <class T>
 inline
-hipError_t hipOccupancyMaxPotentialBlockSize(uint32_t* gridSize, uint32_t* blockSize,
-    F kernel, size_t dynSharedMemPerBlk, uint32_t blockSizeLimit) {
+hipError_t hipOccupancyMaxPotentialBlockSize(int* gridSize, int* blockSize,
```

Comment:
The two template functions in functional_grid_launch.hpp can be removed because  all template functions can be put into include/hip/hcc_detail/hip_runtime_api.h .  Thus   Vdi and Hcc Rt can share the same copy of template functions .
See https://github.com/ROCm-Developer-Tools/HIP/pull/1946/files.

If these are done here,   https://github.com/ROCm-Developer-Tools/HIP/pull/1946  can be abandoned.